dao/mongodb.rya/src/main/java/org/apache/rya/mongodb/dao/SimpleMongoDBStorageStrategy.java

```diff
@@ -64,14 +68,14 @@
     @Override
     public void createIndices(final DBCollection coll){
         BasicDBObject doc = new BasicDBObject();
-        doc.put(SUBJECT, 1);
-        doc.put(PREDICATE, 1);
+        doc.put(SUBJECT_HASH, 1);
+        doc.put(PREDICATE_HASH, 1);
         coll.createIndex(doc);
```

Comment:
When the Mongo db backend was first implemented, you could only do indices over two fields-- the first is the primary index, the second the secondary index.  That may have changed since.  The indices we originally had were subject, predicate, object, and then subject/predicate, predicate/object, and object/subject.  The not including object type might be a bug, but I had thought that was addressed at some point.  Also one could argue that the single field indices were redundant-- I had wanted to test to see but never got around to it.
If you can now index over more than two fields, then we might want to revisit this.